### DISCUSSION TITLE
Angular Velocity PrimeHub

### DISCUSSION THREAD

**Discussion by Redstonetill** [pybricks/support#1735] [Q&A] [answered]
Hey guys, i am using pybricks with a Spike Prime and am looking for a way of tracking the position of my robot. In the documentation (https://docs.pybricks.com/en/stable/hubs/primehub.html) is a function for the angular_velocity() of the hub mentioned. But with this I was wondering which radius I am supposed to use to get the linear velocity. I didnt really find anything about that in the documentation. So I would greatly appreciate if anyone could help.

**Comment by laurensvalk** [comment]
A gyroscope can only measure rotation, not translation.

You could use the distance driven by the wheels to get an estimate of your movements.

**Reply by Redstonetill** [reply]
Ok, but why is there then a function for the angular_velocity()
![Bild_2024-07-29_223704080](https://github.com/user-attachments/assets/ff1633f4-2d5d-4135-9645-2928385ca229)

**Reply by Redstonetill** [reply]
I have also done a little bit more searching and the offical lego "datasheet" (https://assets.education.lego.com/v3/assets/blt293eea581807678a/bltf512a371e82f6420/5f8801baf4f4cf0fa39d2feb/techspecs_techniclargehub.pdf?locale=en-us) says that the spike prime not only has a gyroscope but also a accelerometer. Is there a way inside of pybricks to access that data or how I can "transform" the angular velocity of the accelerometer into linear acceleration?

**Reply by dlech** [reply, marked as answer]
You can get the acceleration with `hub.imu.acceleration()`. https://docs.pybricks.com/en/stable/hubs/primehub.html#pybricks.hubs.PrimeHub.imu.acceleration

But based on my experience, I don't think the sensor is accurate enough and/or can be sampled fast enough to get an accurate velocity from integrating the acceleration over time.

Laurens' suggestion of using the rotation sensors on the motors will be more accurate.

**Reply by Redstonetill** [reply]
Ok thanks, I will try to do that
